Utilities/WDLs/Interval2Bed.wdl

```diff
@@ -0,0 +1,92 @@
+version 1.0
+
+workflow Interval2Bed {
+    input {
+        Array[File] interval_files
+        Array[String]? interval_labels
+    }
+
+    if (defined(interval_labels)) {
+        Array[Pair[String, File]] paired_intervals = zip(select_first([interval_labels]), interval_files)
+
+        scatter (p in paired_intervals) {
+            # Check that file extension is .interval_list, i.e. removing it shortens the basename
+            if (basename(p.right, ".interval_list") != basename(p.right)) {
+                call Convert as ConvertPairs {
+                    input:
+                        interval_list=p.right,
+                        interval_label=p.left
+                }
+            }
+
+            if (basename(p.right, ".bed") != basename(p.right)) {
+                File bed_file_pair = p.right
+                File bed_label_pair = p.left
+            }
+        }
+
+        Array[File] new_interval_files_pair = flatten([select_all(bed_file_pair), select_all(ConvertPairs.bed_file)])
+        Array[File] new_interval_labels_pair = flatten([select_all(bed_label_pair), select_all(ConvertPairs.bed_label)])
+    }
+
+    if (!defined(interval_labels)) {
```

Comment:
I was trying to think through if there was a way to have one block without this `defined` check using a bunch of `select_first(p.left, basename(p.left, "interval_list)` but having the zip makes handling that a bit ugly right? I'm assuming you also thought through that option and concluded that this is cleanest way to do this?

Reply:
Yeah I'm not sure if there are some WDL tricks to make this a bit more elegant, but thought this might be the simplest. I really needed to be able to keep the labels next to the files to ensure they get paired in the outputs as well, and this felt like two different cases. If you have some ideas I'd definitely like to make this nicer!

Reply:
sounds good, leave it as is then. I think all of my ideas just ultimately made things messier.